### ISSUE TITLE
MountVolume.SetUp failed for volume "deployer-conf" : object "default"/"crunchydata-dravoka-postgres-operator-cm" not registered

### ISSUE THREAD

**Issue by rg2609** [kubernetes/kubernetes#110786]
### What happened?

Hi,
I am trying to install the helm chart v4.7.5, but it is giving an error MountVolume.SetUp failed for value 
```
  Type     Reason       Age                From                                      Message
  ----     ------       ----               ----                                      -------
  Normal   Scheduled    <unknown>                                                    Successfully assigned default/pgo-deploy-install--1-qdvj5 to aks-default-12069374-vmss000000
  Normal   Pulling      82s                kubelet, aks-default-12069374-vmss000000  Pulling image "registry.developers.crunchydata.com/crunchydata/pgo-deployer:centos8-4.7.5"
  Normal   Pulled       61s                kubelet, aks-default-12069374-vmss000000  Successfully pulled image "registry.developers.crunchydata.com/crunchydata/pgo-deployer:centos8-4.7.5" in
 21.076257443s
  Normal   Created      60s                kubelet, aks-default-12069374-vmss000000  Created container pgo-deploy
  Normal   Started      60s                kubelet, aks-default-12069374-vmss000000  Started container pgo-deploy
  Warning  FailedMount  34s (x2 over 35s)  kubelet, aks-default-12069374-vmss000000  MountVolume.SetUp failed for volume "deployer-conf" : object "default"/"crunchydata-dravoka-postgres-oper
ator-cm" not registered
```
Values.yaml:
```
---
# ======================
# Installer Controls
# ======================
fullnameOverride: ""

# rbac: settings for deployer RBAC creation
rbac:
    # rbac.create: if false RBAC resources should be in place
    create: true
    # rbac.useClusterAdmin: creates a ClusterRoleBinding giving cluster-admin to serviceAccount.name
    useClusterAdmin: false

# serviceAccount: settings for Service Account used by the deployer
serviceAccount:
    # serviceAccount.create: Whether to create a Service Account or not
    create: true
    # serviceAccount.name: The name of the Service Account to create or use
    name: ""

# =====================
# Configuration Options
# More info for these options can be found in the docs
# https://access.crunchydata.com/documentation/postgres-operator/latest/installation/configuration/
# =====================
archive_mode: "true"
archive_timeout: "60"
backrest_aws_s3_bucket: ""
backrest_aws_s3_endpoint: ""
backrest_aws_s3_key: ""
backrest_aws_s3_region: ""
backrest_aws_s3_secret: ""
backrest_aws_s3_uri_style: ""
backrest_aws_s3_verify_tls: "true"
backrest_gcs_bucket: ""
backrest_gcs_endpoint: ""
backrest_gcs_key_type: ""
backrest_port: "2022"
badger: "false"
ccp_image_prefix: "registry.developers.crunchydata.com/crunchydata"
ccp_image_pull_secret: ""
ccp_image_pull_secret_manifest: ""
ccp_image_tag: "centos8-13.6-4.7.5"
create_rbac: "true"
crunchy_debug: "false"
db_name: "postgres"
db_password_age_days: "0"
db_password_length: "24"
db_port: "5432"
db_replicas: "0"
db_user: "uasername"
default_instance_memory: "128Mi"
default_pgbackrest_memory: "48Mi"
default_pgbouncer_memory: "24Mi"
default_exporter_memory: "24Mi"
delete_operator_namespace: "false"
delete_watched_namespaces: "false"
disable_auto_failover: "false"
disable_fsgroup: "false"
reconcile_rbac: "true"
exporterport: "9187"
metrics: "false"
namespace: "pgo"
namespace_mode: "dynamic"
pgbadgerport: "10000"
pgo_add_os_ca_store: "false"
pgo_admin_password: "password"
pgo_admin_perms: "*"
pgo_admin_role_name: "pgoadmin"
pgo_admin_username: "admin"
pgo_apiserver_port: "8443"
pgo_apiserver_url: "https://postgres-operator"
pgo_client_cert_secret: "pgo.tls"
pgo_client_container_install: "true"
pgo_client_install: "true"
pgo_client_version: "4.7.5"
pgo_cluster_admin: "false"
pgo_disable_eventing: "false"
pgo_disable_tls: "false"
pgo_image_prefix: "registry.developers.crunchydata.com/crunchydata"
pgo_image_pull_secret: ""
pgo_image_pull_secret_manifest: ""
pgo_image_tag: "centos8-4.7.5"
pgo_installation_name: "devtest"
pgo_noauth_routes: ""
pgo_operator_namespace: "pgo"
pgo_tls_ca_store: ""
pgo_tls_no_verify: "false"
pod_anti_affinity: "preferred"
pod_anti_affinity_pgbackrest: ""
pod_anti_affinity_pgbouncer: ""
scheduler_timeout: "3600"
service_type: "ClusterIP"
sync_replication: "false"
backrest_storage: "hostpathstorage"
backup_storage: "hostpathstorage"
primary_storage: "hostpathstorage"
replica_storage: "hostpathstorage"
pgadmin_storage: "hostpathstorage"
wal_storage: ""
storage1_name: "hostpathstorage"
storage1_access_mode: "ReadWriteMany"
storage1_size: "1G"
storage1_type: "create"
storage2_name: "replicastorage"
storage2_access_mode: "ReadWriteMany"
storage2_size: "1G"
storage2_type: "create"
storage3_name: "nfsstorage"
storage3_access_mode: "ReadWriteMany"
storage3_size: "6G"
storage3_type: "dynamic"
storage3_class: "nfs"
storage3_supplemental_groups: "65534"
storage4_name: "nfsstoragered"
storage4_access_mode: "ReadWriteMany"
storage4_size: "1G"
storage4_match_labels: "crunchyzone=red"
storage4_type: "create"
storage4_supplemental_groups: "65534"
storage5_name: "storageos"
storage5_access_mode: "ReadWriteOnce"
storage5_size: "5Gi"
storage5_type: "dynamic"
storage5_class: "fast"
storage6_name: "primarysite"
storage6_access_mode: "ReadWriteOnce"
storage6_size: "4G"
storage6_type: "dynamic"
storage6_class: "primarysite"
storage7_name: "alternatesite"
storage7_access_mode: "ReadWriteOnce"
storage7_size: "4G"
storage7_type: "dynamic"
storage7_class: "alternatesite"
storage8_name: "gce"
storage8_access_mode: "ReadWriteOnce"
storage8_size: "300M"
storage8_type: "dynamic"
storage8_class: "standard"
storage9_name: "rook"
storage9_access_mode: "ReadWriteOnce"
storage9_size: "1Gi"
storage9_type: "dynamic"
storage9_class: "rook-ceph-block"
```

### What did you expect to happen?

It should install the pgo

### How can we reproduce it (as minimally and precisely as possible)?

install this [https://github.com/CrunchyData/postgres-operator/tree/v4.7.5/installers/helm]( https://github.com/CrunchyData/postgres-operator/tree/v4.7.5/installers/helm) with nfs storage 

### Kubernetes version

<details>

```console
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.3", GitCommit:"2e7996e3e2712684bc73f0dec0200d64eec7fe40", GitTreeState:"clean", BuildDate:"2020-05-20T12:52:00Z", GoVers
ion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.6", GitCommit:"07959215dd83b4ae6317b33c824f845abd578642", GitTreeState:"clean", BuildDate:"2022-03-30T18:28:25Z", GoVers
ion:"go1.16.12", Compiler:"gc", Platform:"linux/amd64"}
```
</details>

### Cloud provider

<details>
Azure
</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
NAME="Ubuntu"
VERSION="16.04.7 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.7 LTS"
VERSION_ID="16.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
VERSION_CODENAME=xenial
UBUNTU_CODENAME=xenial
$ uname -a
# Linux vm-bastion 4.15.0-1113-azure #126~16.04.1-Ubuntu SMP Tue Apr 13 16:55:24 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by rg2609** [comment]
also tried steps from [here](https://www.crunchydata.com/developers/download-postgres/containers/postgres-operator)
```console
kubectl create namespace pgo
kubectl apply -f https://raw.githubusercontent.com/CrunchyData/postgres-operator/v4.7.5/installers/kubectl/postgres-operator.yml
```
but this also giving en error

```console
abc_user@vm-bastion:~$ kubectl get pod -n pgo
NAME                  READY   STATUS   RESTARTS   AGE
pgo-deploy--1-8rrl5   0/1     Error    0          25s
dravoka_user@vm-bastion:~$ kubectl describe pod pgo-deploy--1-8rrl5 -n pgo
Name:         pgo-deploy--1-8rrl5
Namespace:    pgo
Priority:     0
Node:         aks-default-12069374-vmss000000/10.1.0.5
Start Time:   Sat, 25 Jun 2022 12:25:23 +0000
Labels:       controller-uid=b7db596f-4fb1-4720-b242-587843d5307f
              job-name=pgo-deploy
              vendor=crunchydata
Annotations:  <none>
Status:       Failed
IP:           172.16.1.35
IPs:
  IP:           172.16.1.35
Controlled By:  Job/pgo-deploy
Containers:
  pgo-deploy:
    Container ID:   containerd://f7204c892334f02c407f3371aba0a714cee832d323e9270e14f66e9e395d4329
    Image:          registry.developers.crunchydata.com/crunchydata/pgo-deployer:centos8-4.7.5
    Image ID:       registry.developers.crunchydata.com/crunchydata/pgo-deployer@sha256:bdd0e1bee68ba0486f611931b63a42cde26abe448dfd46ddb3f32dd4131ec55b
    Port:           <none>
    Host Port:      <none>
    State:          Terminated
      Reason:       Error
      Exit Code:    2
      Started:      Sat, 25 Jun 2022 12:25:24 +0000
      Finished:     Sat, 25 Jun 2022 12:25:31 +0000
    Ready:          False
    Restart Count:  0
    Environment:
      DEPLOY_ACTION:  install
    Mounts:
      /conf from deployer-conf (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-qqqpb (ro)
Conditions:
  Type              Status
  Initialized       True 
  Ready             False 
  ContainersReady   False 
  PodScheduled      True 
Volumes:
  deployer-conf:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      pgo-deployer-cm
    Optional:  false
  kube-api-access-qqqpb:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:                   BestEffort
Node-Selectors:              <none>
Tolerations:                 node.kubernetes.io/not-ready:NoExecute for 300s
                             node.kubernetes.io/unreachable:NoExecute for 300s
Events:
  Type     Reason       Age                From                                      Message
  ----     ------       ----               ----                                      -------
  Normal   Scheduled    <unknown>                                                    Successfully assigned pgo/pgo-deploy--1-8rrl5 to aks-default-12069374-vmss000000
  Normal   Pulled       60s                kubelet, aks-default-12069374-vmss000000  Container image "registry.developers.crunchydata.com/crunchydata/pgo-deployer:centos8-4.7.5" already pres
ent on machine
  Normal   Created      60s                kubelet, aks-default-12069374-vmss000000  Created container pgo-deploy
  Normal   Started      60s                kubelet, aks-default-12069374-vmss000000  Started container pgo-deploy
  Warning  FailedMount  52s (x2 over 52s)  kubelet, aks-default-12069374-vmss000000  MountVolume.SetUp failed for volume "kube-api-access-qqqpb" : object "pgo"/"kube-root-ca.crt" not registe
red
  Warning  FailedMount  52s (x2 over 52s)  kubelet, aks-default-12069374-vmss000000  MountVolume.SetUp failed for volume "deployer-conf" : object "pgo"/"pgo-deployer-cm" not registered
```

**Comment by liggitt** [comment]
this looks like the same issue as https://github.com/kubernetes/kubernetes/issues/105204#issuecomment-1104427967, resolved in 1.22.9+ and 1.23.6+, and 1.24+

**Comment by rg2609** [comment]
I took the takes of 1.24.2 but still it failed
